src/core/mac/sub_mac.cpp

```diff
@@ -396,12 +396,13 @@ void SubMac::StartCsmaBackoff(void)
         {
             static constexpr uint32_t kAheadTime = kCcaSampleInterval + kCslTransmitTimeAhead + kRadioHeaderShrDuration;
             Time                      txStartTime = Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime);
+            Time                      now         = Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance())));
 
             txStartTime += (mTransmitFrame.mInfo.mTxInfo.mTxDelay - kAheadTime);
 
-            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) < txStartTime)
+            if (now < txStartTime)
             {
-                mTimer.FireAt(txStartTime);
+                StartTimer(txStartTime.GetValue() - now.GetValue());
```

Comment:
I think we can do this:
```suggestion
                StartTimer(txStartTime - now);
```
 (can subtract to `Time` instances which give us the duration between them).